composer/callbacks/memory_monitor.py

```diff
@@ -76,12 +76,11 @@ class MemoryMonitor(Callback):
         Memory usage monitoring is only supported for the GPU devices.
     """
 
-    def __init__(self):
-        super().__init__()
-        log.info(
-            "Memory monitor just profiles the current GPU assuming that the memory footprint across GPUs is balanced.")
-        if torch.cuda.device_count() == 0:
-            log.warning("Memory monitor only works on GPU devices.")
+    def init(self, state: State, logger: Logger) -> None:
+        model_device = next(state.model.parameters()).device
+
+        if model_device.type != 'cuda':
+            raise RuntimeError("The memory monitor only works on CUDA devices, but the model is on CPU.")
```

Comment:
I don't like throwing an error here. Now a user will specifically have to delete this callback or their run still stop if they use the same yaml or script on cpu. I'd prefer to disable it for them in that case rather than forcing code/config chances.